R/dominantTaxa.R

```diff
@@ -100,9 +114,50 @@ setMethod("perSampleDominantFeatures", signature = c(x = "SummarizedExperiment")
         idx <- as.list(apply(t(mat) == colMaxs(mat),1L,which))
         # Get rownames based on indices
         taxa <- rownames(mat)[unlist(idx)]
+        
         # If multiple dominant taxa were found, names contain taxa in addition to 
         # sample name. Names are converted so that they include only sample names.
         names(taxa) <- rep( names(idx), times = lengths(idx) )
+        
+        # If individual sample contains multiple dominant taxa (they have equal counts) and if 
+        # complete is FALSE, the an arbitrarily chosen dominant taxa is returned
+        if( length(taxa)>ncol(x) && !complete){
+            # Store order
+            order <- unique(names(taxa))
+            # there are multiple dominant taxa in one sample (counts are equal), length
+            # of dominant is greater than rows in colData.
+            taxa <- split(taxa, rep(names(taxa), lengths(taxa)) )
+            # Order the data
+            taxa <- taxa[order]
+            names <- names(taxa)
+            # If complete is set FALSE, and there are multiple dominant taxa, one of them is arbitrarily chosen
+            taxa <- lapply(taxa, function(item) {
+                        return(sample(item, 1)) })
+            taxa <- unname(sapply(taxa, function (x) {
+                        unlist(x)}))
+            names(taxa) <- names
+            warning("Multiple dominant taxa were found for some samples. Run perSampleDominantFeatures(x, complete = TRUE) for details.")
```

Comment:
As this is also used in add* function, the message is little bit unclear --> it might also be "Multiple dominant taxa were found for some samples. Use complete = TRUE for details."